tests/cxx/elaborated_type.cc

```diff
@@ -60,6 +60,21 @@ void global_qualified_elab(class ::GlobalClass* c);
 // they must also be forward-declared.
 Elaboration::Template<int, float>* namespace_qualified_template;
 
+// Elaborated types can be treated as forward-declarations, hence no fwd-decl is
+// needed if a declaration is introduced by elaborated type.
+// A declaration "owned" by elaborated type is an implicit declaration
+// introduced by elaborated type if no one was already present.
+void owning_elaborated(class OwningElaborated*);
```

Comment:
It's curious that this declaration "falls out" into the containing scope/decl context. Are the rules different between C and C++ here? https://github.com/include-what-you-use/include-what-you-use/blob/master/iwyu.cc#L4171.

Reply:
That C language rule seems strange, as for me... But with the current implementation, specific visibility rules don't matter, I think. An elaborated type does or does not introduce a new "owned" declaration. Later usages refer to some declaration. IWYU matches usages to declarations, when all the lookup work should already be done by clang.

Reply:
Btw, I thought whether I might break that case with C function, but it luckily seems that I haven't. The behavior seems to be the same as on the master branch.

Reply:
Yeah, there are tests for that under `tests/c`, so we should be good :)